### DISCUSSION TITLE
Documentation on Themes

### DISCUSSION THREAD

**Discussion by louhy** [AvaloniaUI/Avalonia#9201] [General]
I'm having a heck of a time finding where to read about switching between Light/Dark themes, maybe somebody can point me in the right direction. Basically I need to be able to switch FluentTheme between Light & Dark at runtime, _the correct way_. I did "discover" a method of doing this that works fine, but it's so awkward (and possibly fragile) I don't think it's the correct way. 

```
        public void ToggleTheme()
        {
            if (Application.Current == null)
                return;

            FluentTheme fluentTheme = GetCurrentFluentTheme();
            FluentThemeMode newFluentThemeMode =
                fluentTheme.Mode == FluentThemeMode.Dark ? FluentThemeMode.Light : FluentThemeMode.Dark;
            SetTheme(newFluentThemeMode);
        }

        //I would really prefer a better way of getting the current FluentTheme
        private FluentTheme GetCurrentFluentTheme()
        {
            foreach (IStyle style in Application.Current.Styles)
            {
                if (!(style is FluentTheme)) continue;

                FluentTheme theme = (FluentTheme)style;
                return theme;
            }

            throw new Exception("Unable to locate FluentTheme");
        }

        public void SetTheme(FluentThemeMode fluentThemeMode)
        {
            FluentTheme fluentTheme = GetCurrentFluentTheme();
            fluentTheme.Mode = fluentThemeMode;
        }
```

Yuck on that foreach loop - the context for this is inside of a class extending ViewModelBase. Any better way? 

Bonus question - 
I am coming from a Java background so this might sound a little clueless and frustrated but I'll work it out eventually. While looking for a solution I came across the post below. 

https://github.com/AvaloniaUI/Avalonia/discussions/5347 

I see there's this "Settings.Instance.ApplicationTheme" _thing_ that's being accessed to change the theme mode. One of the frustrating things about C# vs Java is the C# tendency to capitalize things (i.e. methods) like crazy instead of only types (class names), so I'm not always clear on what the heck I'm looking at. In Java this would be a reference to a static method (and generally a frowned upon practice) since Settings would be a class, being capitalized. 

So is this "Settings" some field that the author of the post just didn't include? Some kind of global static thingy that should be available for me to access? My guess is it's some global thing, but it doesn't appear to be available in the class I'm modifying. It probably is clear what's going on, and I'm just not used to it...

**Comment by sn4k3** [comment]
If you are using Avalonia 0.10.x then this is one way: https://github.com/sn4k3/UVtools/blob/693188b961d5e7acaf3969b72264b72483d4c33d/UVtools.WPF/App.axaml.cs#L70-L179

For 0.11.x i have no clue if is like the same.

> //I would really prefer a better way of getting the current FluentTheme

You can do like i did, create and insert FluentTheme variable, then you always have it variable to manipulate instead of foreach each style

https://github.com/sn4k3/UVtools/blob/693188b961d5e7acaf3969b72264b72483d4c33d/UVtools.WPF/App.axaml.cs#L90
https://github.com/sn4k3/UVtools/blob/693188b961d5e7acaf3969b72264b72483d4c33d/UVtools.WPF/App.axaml.cs#L181-L183

But i don't see any problem with your code. FluentTheme style can't gone by magic
